squad/ci/backend/lava.py

```diff
@@ -60,19 +60,18 @@ def fetch(self, test_job):
             return self.__parse_results__(data)
 
     def listen(self):
-        if not self.data.listener_url:
-            self.log_debug("no listener_url, cannot listen for events")
-            return
+        url = urlsplit(self.data.url)
+        hostname = url.netloc
+
+        socket = self.__get_publisher_event_socket__()
```

Comment:
I only checked with LKFT and it returned the proper address :( Anyway, I think your solution is OK. Just need to take the protocol/scheme from the reply to form the final address.